app/models/storage.rb

```diff
@@ -70,6 +71,12 @@ class Storage < ApplicationRecord
 
   SUPPORTED_STORAGE_TYPES = %w( VMFS NFS FCP ISCSI GLUSTERFS )
 
+  supports :smartstate_analysis do
+    if ext_management_systems.blank? || !ext_management_system.kind_of?(ManageIQ::Providers::Vmware::InfraManager)
```

Comment:
@gauravaradhye as per the linked gitter discussion above, please refactor this to read
```ruby
if ext_management_systems.empty? || !ext_management_system.class.supports_storage_smartstate_analysis?
```
Then default `supports_storage_smartstate_analysis` to no on Providers::BaseManager and change to supports for Vmware::Inframanager